### ISSUE TITLE
iron-list gives error when the objects have a field named "children" in them. 

### ISSUE THREAD

**Issue by satyanath** [PolymerElements/iron-list#351]
### Description

iron-list element gives an error when the list of objects passed to it has the field "children" in the objects. It gives an "Illegal Invocation" exception. 

I use google chrome browser on windowws 10 - version - 54.0.2840.87 m

### Expected outcome

The exception should not occur and the page should display

Hawkins Parrish
Dena Sweeney
Saundra Mcbride

### Actual outcome

Exception is raised.
polymer.html:2053 Uncaught TypeError: Illegal invocation(…) 
in the debug console.

### Live Demo

html file
-------------
<!DOCTYPE html>
<html>

<head>
    <title>Iron list test</title>
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <script src="bower_components/webcomponentsjs/webcomponents.min.js"></script>
    <link rel="import" href="bower_components/polymer/polymer.html">
    <link rel="import" href="bower_components/iron-ajax/iron-ajax.html">
    <link rel="import" href="bower_components/iron-list/iron-list.html">

</head>

<body>
  <template is="dom-bind">
    <iron-ajax url="data2.json" last-response="{{data}}" auto></iron-ajax>
    <iron-list items="[[data]]" as="item">
        <template>
            <div>[[item.posttext]]</div>
        </template>
    </iron-list>
  </template>
</body>

</html>

data2.json file
-----------------
[
  {
    "posttext": "Hawkins Parrish",
    "havechildren": "true",
    "children": "dddd"
  },
  {
    "posttext": "Dena Sweeney",
    "havechildren": "true",
    "children": "dddd"
  },
  {
    "posttext": "Saundra Mcbride",
    "havechildren": "true",
    "children": "dddd"
  }
]

### Browsers Affected

- [ X] Chrome
- [X ] Firefox
- [ ] Safari 9
- [ ] Safari 8
- [ ] Safari 7
- [X ] Edge
- [ ] IE 11
- [ ] IE 10

Not checked on others
[PolymerTest.zip](https://github.com/PolymerElements/iron-list/files/585050/PolymerTest.zip)

**Comment by blasten** [comment]
Yep, I have seen this issue in the past. This is because it's overriding `Node.children`. Unfortunately, this is a limitation of v1, probably solved in v2.

**Comment by satyanath** [comment]
OK then.

Regards,

S.Satyanath
